eagle-server/src/main/java/org/apache/eagle/server/ServerApplication.java

```diff
@@ -107,18 +111,31 @@ public void run(ServerConfig configuration, Environment environment) throws Exce
         // Context listener
         environment.servlets().addServletListeners(new CoordinatorListener());
 
+        registerAppServices(environment);
+    }
+
+    private void registerAppServices(Environment environment) {
         // Run application status service in background
-        Managed updateAppStatusTask = new ApplicationTask(applicationStatusUpdateService);
+        LOG.debug("Registering ApplicationStatusUpdateService");
+        Managed updateAppStatusTask = new ManagedService(applicationStatusUpdateService);
         environment.lifecycle().manage(updateAppStatusTask);
 
-        // Initialize application health check environment
+        // Initialize application extended health checks.
+        LOG.debug("Registering ApplicationHealthCheckService");
         applicationHealthCheckService.init(environment);
-        Managed appHealthCheckTask = new ApplicationTask(applicationHealthCheckService);
-        environment.lifecycle().manage(appHealthCheckTask);
-
-        if (config.hasPath(MRHistoryJobDailyReporter.SERVICE_PATH)) {
-            Managed jobReportTask = new ApplicationTask(mrHistoryJobDailyReporter);
-            environment.lifecycle().manage(jobReportTask);
+        environment.lifecycle().manage(new ManagedService(applicationHealthCheckService));
+
+        // Load application shared extension services.
+        LOG.debug("Registering application shared extension services");
+        for (ApplicationProvider<?> applicationProvider : applicationProviderService.getProviders()) {

```

Comment:
Change#3, Register `getSharedServices` from `ApplicationProvider` to `Environment` in `ServerApplication`